ckanext/switzerland/helpers/format_utils.py

```diff
@@ -32,59 +32,51 @@ def ogdch_get_format_mapping():
         return format_mapping
 
 
-# Generates format of resource and saves it in format field
-def prepare_resource_format(resource, format_mapping):
-    resource_format = ''
-
-    # get format from media_type field if available
-    if not resource_format and resource.get('media_type'):  # noqa
-        resource_format = resource['media_type'].split('/')[-1].lower()
-
-    # get format from format field if available (lol)
-    if not resource_format and resource.get('format'):
-        resource_format = resource['format'].split('/')[-1].lower()
-
-    # check if 'media_type' or 'format' can be mapped
-    has_format = (_map_to_valid_format(
-        resource_format,
-        format_mapping
-    ) is not None)
-
-    # if the fields can't be mapped,
-    # try to parse the download_url as a last resort
-    if not has_format and resource.get('download_url'):
-        path = urlparse.urlparse(resource['download_url']).path
-        ext = os.path.splitext(path)[1]
-        if ext:
-            resource_format = ext.replace('.', '').lower()
-
-    mapped_format = _map_to_valid_format(
-        resource_format,
-        format_mapping
-    )
-    if mapped_format:
-        # if format could be successfully mapped write it to format field
-        resource['format'] = mapped_format
-    elif not resource.get('download_url'):
-        resource['format'] = 'SERVICE'
-    else:
-        # else return empty string (this will be indexed as N/A)
-        resource['format'] = ''
-
-    # if format could not be mapped and media_type exists use this value  # noqa
-    if (not resource.get('format') and resource.get('media_type')):
-        resource['format'] = resource['media_type'].split('/')[-1]
+format_mapping = ogdch_get_format_mapping()
+
+
+def prepare_resource_format(resource):
```

Comment:
This function has been refactored to make the 4 different cases easier to understand. Because tests for this functions were already there and a new test has been added, I felt it was safe to do this and will make future work on the format handeling a lot easier.